app/models/gem_info.rb

```diff
@@ -105,6 +105,7 @@ def compute_compact_index_info
           deps << CompactIndex::Dependency.new(name, req) unless name == '0'
         end
       end
+      [r[4], r[5]].each { |req| req&.gsub!(/,\s*/, '&') }
```

Comment:
I think that we should do this in https://github.com/rubygems/compact_index/blob/426b4bba20f84fff523f7901a27b7b09eba05285/lib/compact_index/gem_version.rb#L27-L28 but the repository is archived...